nussl/separation/duet.py

```diff
@@ -113,102 +164,78 @@ def run(self):
-
         # Save these for later
-        self.attenuation_grid = agrid
-        self.delay_grid = dgrid
-        self.attenuation_delay_histogram = hist
-        self.non_normalized_hist = H[0]
-        self.smoothed_hist = self._smooth_matrix(self.non_normalized_hist, np.array([3]))
+        self.attenuation_grid = atn_bins
+        self.delay_grid = delay_bins
 
-        # smooth the histogram - local average 3-by-3 neighboring bins
-        hist = self._smooth_matrix(hist, np.array([3]))
+        # TODO: do these operations happen twice?
+        # resolved
+        self.attenuation_delay_histogram = hist / hist.max()
+        self.non_normalized_hist = hist
+
+        # smooth the normalized histogram - local average 3-by-3 neighboring bins
+        hist = self._smooth_matrix(self.attenuation_delay_histogram, np.array([3]))
 
         # normalize and plot the histogram
         hist /= hist.max()
 
-        # find the location of peaks in the alpha-delta plane
-        self.peak_indices = self.find_peaks2(hist, self.peak_threshold,
-                                             np.array([self.attenuation_min_distance, self.delay_min_distance]),
-                                             self.num_sources)
-
-        alphapeak = agrid[self.peak_indices[0, :]]
-        deltapeak = dgrid[self.peak_indices[1, :]]
-
-        ad_est = np.vstack([alphapeak, deltapeak]).T
-
-        # convert alpha to a
-        atnpeak = (alphapeak + np.sqrt(alphapeak ** 2 + 4)) / 2
-
-        # compute masks for separation
-        bestsofar = np.inf * np.ones((num_frequency_bins - 1, num_time_bins))
-        bestind = np.zeros((num_frequency_bins - 1, num_time_bins), int)
-        for i in range(0, self.num_sources):
-            score = np.abs(atnpeak[i] * np.exp(-1j * frequency_matrix * deltapeak[i]) * stft_ch0 - stft_ch1) ** 2 / \
-                    (1 + atnpeak[i] ** 2)
-            mask = (score < bestsofar)
-            bestind[mask] = i
-            bestsofar[mask] = score[mask]
-
-        # demix with ML alignment and convert to time domain
-        xhat = np.zeros((self.num_sources, self.audio_signal.signal_length))
-        for i in range(0, self.num_sources):
-            mask = (bestind == i)
-            Xm = np.vstack([np.zeros((1, num_time_bins)),
-                            (stft_ch0 + atnpeak[i] * np.exp(1j * frequency_matrix * deltapeak[i]) * stft_ch1) /
-                            (1 + atnpeak[i] ** 2) * mask])
-            # xi = spectral_utils.f_istft(Xm, L, winType, hop, fs)
-            xi = nussl.spectral_utils.e_istft(Xm, L, hop, win_type)
-
-            xhat[i, :] = nussl.utils.add_mismatched_arrays(xhat[i,], xi)[:self.audio_signal.signal_length]
-            # add back to the separated signal a portion of the mixture to eliminate
-            # most of the masking artifacts
-            # xhat=xhat+0.05*x[0,:]
-
-        self.separated_sources = xhat
-
-        return xhat, ad_est
+        return hist, atn_bins, delay_bins
```

Comment:
Do you need to return these things if they're already saved as attributes? Do you use the attributes or do you use the return values? Pick one.